### PR TITLE
feat(tour): tour support custom style and fill color

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- feat: #39780 

### 💡 Background and solution

- support pass custom style and fill color to change tour's mask style

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  support pass custom style and fill color to change tour's mask style  |
| 🇨🇳 Chinese | 支持传递自定义样式和填充颜色改变 tour 的遮罩样式  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

### CODE REVIEW

File: components/tour/index.en-US.md [hunk 1]
```diff
@@ -30,7 +31,7 @@ Use when you want to guide users through a product.
 | arrow | Whether to show the arrow, including the configuration whether to point to the center of the element | `boolean`\|`{ pointAtCenter: boolean}` | `true` |  |
 | placement | Position of the guide card relative to the target element | `left` `leftTop` `leftBottom` `right` `rightTop` `rightBottom` `top` `topLeft` `topRight` `bottom` `bottomLeft` `bottomRight` | `bottom` |  |
 | onClose | Callback function on shutdown | `Function` | - |  |
-| mask | Whether to enable masking | `boolean` | `true` |  |
+| mask | Whether to enable masking，change mask style and fill color by pass custom props | `boolean \| { style?: React.CSSProperties; fill?: string; }` | `true` |  |
```
Comment (afc163) [inline components/tour/index.en-US.md:34]:
```suggestion
| mask | Whether to enable masking, change mask style and fill color by pass custom props | `boolean \| { style?: React.CSSProperties; fill?: string; }` | `true` |  |
```

File: components/tour/index.en-US.md [hunk 2]
```diff
@@ -47,7 +48,7 @@ Use when you want to guide users through a product.
 | description | description | `ReactNode` | - |  |
 | placement | Position of the guide card relative to the target element | `left` `leftTop` `leftBottom` `right` `rightTop` `rightBottom` `top` `topLeft` `topRight` `bottom` `bottomLeft` `bottomRight` | `bottom` |  |
 | onClose | Callback function on shutdown | `Function` | - |  |
-| mask | Whether to enable masking, the default follows the `mask` property of Tour | `boolean` | `true` |  |
+| mask | Whether to enable masking，change mask style and fill color by pass custom props, the default follows the `mask` property of Tour | `boolean \| { style?: React.CSSProperties; fill?: string; }` | `true` |  |
```
Comment (afc163) [inline components/tour/index.en-US.md:51]:
```suggestion
| mask | Whether to enable masking, change mask style and fill color by pass custom props, the default follows the `mask` property of Tour | `boolean \| { style?: React.CSSProperties; fill?: string; }` | `true` |  |
```

File: components/tour/index.zh-CN.md [hunk 3]
```diff
@@ -18,6 +18,7 @@ demo:
 
 <!-- prettier-ignore -->
 <code src="./demo/basic.tsx">基本</code>
+<code src="./demo/customMask.tsx">自定义遮罩样式</code>
```
Comment (afc163) [inline components/tour/index.zh-CN.md:21]: 这个例子放后面

Comment (afc163) [reply components/tour/index.zh-CN.md:21]: mask.tsx 就行了

File: components/tour/demo/customMask.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    },
+    {
+      title: 'Save',
+      description: 'Save your changes.',
+      target: () => ref2.current,
+      mask: {
+        style: {
+          boxShadow: 'inset 0 0 15px #fff',
+        },
+        fill: 'rgba(40, 0, 255, .4)',
```
Comment (zombieJ) [inline components/tour/demo/customMask.tsx:33]: 有办法和 `style` 合并起来么？这个 svg 属性有点蛋疼。

Comment (kiner-tang) [reply components/tour/demo/customMask.tsx:33]: 这个不太好合，毕竟`style`得属性就那么些，放在`style`感觉有点违背`style`类型，用户用起来也会很奇怪。

File: components/tour/index.zh-CN.md (hunk 3 above)
Comment (kiner-tang) [reply components/tour/index.zh-CN.md]: done

File: components/tour/demo/mask.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    },
+    {
+      title: 'Save',
+      description: 'Save your changes.',
+      target: () => ref2.current,
+      mask: {
+        style: {
+          boxShadow: 'inset 0 0 15px #fff',
+        },
+        fill: 'rgba(40, 0, 255, .4)',
```
Comment (zombieJ) [inline components/tour/demo/mask.tsx:33]: 主要这个 `fill` 过于裸了，如果未来重构了这部分不是 svg 实现的。`fill` 就成为坏味道了。

Comment (zombieJ) [reply components/tour/demo/mask.tsx:33]: 我会倾向于就提供一个 `mask.color` 然后转给 `fill` 上隔离底层的 svg

Comment (kiner-tang) [reply components/tour/demo/mask.tsx:33]: 也可以

Comment (kiner-tang) [reply components/tour/demo/mask.tsx:33]: 那干脆`rc-tour`也改一下api，直接用`color`把，省得再`ant-design`还要转来转去的 @zombieJ

**Comment by afc163** [comment]
feature branch

**Comment by kiner-tang** [comment]
> feature branch

ok

File: components/tour/demo/mask.tsx (hunk 5 above)
Comment (kiner-tang) [reply components/tour/demo/mask.tsx:33]: @zombieJ 帮忙看一下这个`pr`,然后我再重新提一个`pr`合到`feature`
- https://github.com/react-component/tour/pull/25

**Comment by kiner-tang** [comment]
see here: https://github.com/ant-design/ant-design/pull/39919

**Pull request closed without merging**